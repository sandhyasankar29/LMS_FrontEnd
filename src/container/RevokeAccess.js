import { React, useState } from 'react';
import axios from 'axios';
import { Box, Grommet, Button, Image, Header, Nav, Anchor, Select, TextInput } from 'grommet';
import { grommet } from 'grommet/themes';
import { User } from 'grommet-icons';
import passImage from '../assets/lock.svg'
import LogOut from './LogOut';
import Swal from 'sweetalert2'
import './GlobalVariables'
export const RevokeAccess = () => {
    const urlRevoke = 'http://'+{global,ip}.ip+':3001/revoke';
    const Toast = Swal.mixin({
        toast: true,
        position: 'top-end',
        showConfirmButton: false,
        timer: 3000,
        timerProgressBar: true,
        didOpen: (toast) => {
            toast.addEventListener('mouseenter', Swal.stopTimer)
            toast.addEventListener('mouseleave', Swal.resumeTimer)
        }
    })
    const [id, setId] = useState('')
    const [userid, setUserId] = useState()
    const items = [
        { label: 'Verify Teacher', href: '/adminteacherverify', value: 0 },
        { label: 'Revoke Access', href: '/adminrevoke', value: 1 },
        { label: 'Button', href: '#', value: 2 },

    ];
    const revoke = () => {
        if (id && userid) {
            axios.post(urlRevoke, { id: id, userid: userid })
                .then(res => {
                    Toast.fire({
                        icon: 'success',
                        title: 'Success'
                    })
                }).catch(error => {
                    Toast.fire({
                        icon: 'error',
                        title: 'Failed'
                    })
                })
        }
    }
    const options = ['student', 'teacher']
    return (
        <Grommet full theme={grommet}>
            <Header background="dark-1" pad="small">
                <Box direction="row" align="center" gap="small">
                    <User />
                    <Anchor color="white" >
                        Welcome Admin
                </Anchor>
                </Box>
                <Nav direction="row">
                    {items.map(item => (
                        item.value !== 2 ? <Anchor href={item.href} label={item.label} key={item.label} /> : <LogOut route={'/adminsignin'} />
                    ))}
                </Nav>

            </Header>
            <Box fill align="center" justify="center">
                <Box height="small" width="large" style={{
                    marginBottom: "3%",
                }}>
                    <Image src={passImage} fit="contain" />
                </Box>

                <Box width="medium">
                    <Box
                        width="large"
                        direction="row"
                        align="center"
                        round="small"
                        border
                    >
                        <TextInput
                            plain
                            placeholder="Enter Registered ID"
                            name="userid" type="text"
                            //eslint-disable-next-line react/jsx-no-duplicate-props
                            type={'text'}
                            value={userid}
                            onChange={(event) => setUserId(event.target.value)}
                        />
                        <Select
                            id="select"
                            name="id"
                            placeholder="Choose.."
                            value={id}
                            options={options}
                            onChange={({ option }) => setId(option)}
                        />
                    </Box>

                    <Box direction="row" justify="center" margin={{ top: 'medium' }}>
                        <Button active={true} onClick={revoke} type="submit" label="Revoke" primary />
                    </Box>
                </Box>
            </Box>
        </Grommet>)
}

export default RevokeAccess;