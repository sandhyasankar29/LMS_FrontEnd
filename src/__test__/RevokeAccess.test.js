import React from 'react';
import ReactDOM from 'react-dom';
import RevokeAccess from '../container/RevokeAccess';
import { cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

afterEach(cleanup)
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<RevokeAccess></RevokeAccess>, div)
})

it("matches snapshot", () => {
    const tree = renderer.create(<RevokeAccess></RevokeAccess>).toJSON();
    expect(tree).toMatchSnapshot();
})
