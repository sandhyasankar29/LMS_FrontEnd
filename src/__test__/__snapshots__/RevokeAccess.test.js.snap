// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 bWDPZI"
>
  <header
    className="StyledBox-sc-13pk1d4-0 jwhdBG"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 dfYOnT"
    >
      <svg
        aria-label="User"
        className="StyledIcon-ofa7kd-0 dcacLZ"
        viewBox="0 0 24 24"
      >
        <path
          d="M8,24 L8,19 M16,24 L16,19 M3,24 L3,19 C3,14.0294373 7.02943725,11 12,11 C16.9705627,11 21,14.0294373 21,19 L21,24 M12,11 C14.7614237,11 17,8.76142375 17,6 C17,3.23857625 14.7614237,1 12,1 C9.23857625,1 7,3.23857625 7,6 C7,8.76142375 9.23857625,11 12,11 Z"
          fill="none"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 eBVJAG"
      />
      <a
        className="StyledAnchor-sc-1rp7lwl-0 ePWGdF"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Welcome Admin
      </a>
    </div>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffHQai"
    />
    <nav
      className="StyledBox-sc-13pk1d4-0 glIhNO"
    >
      <a
        className="StyledAnchor-sc-1rp7lwl-0 eaoGTv"
        href="/adminteacherverify"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Verify Teacher
      </a>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffHQai"
      />
      <a
        className="StyledAnchor-sc-1rp7lwl-0 eaoGTv"
        href="/adminrevoke"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Revoke Access
      </a>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffHQai"
      />
      <button
        className="StyledButton-sc-323bzc-0 bZMlsb"
        id="logoutBtn"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        <svg
          aria-label="Logout"
          className="StyledIcon-ofa7kd-0 eBBIEF"
          viewBox="0 0 24 24"
        >
          <path
            d="M13,9 L13,2 L1,2 L1,22 L13,22 L13,15 M22,12 L5,12 M17,7 L22,12 L17,17"
            fill="none"
            stroke="#231F20"
            strokeWidth="2"
          />
        </svg>
      </button>
    </nav>
  </header>
  <div
    className="StyledBox-sc-13pk1d4-0 bxevCE"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 bJtzEt"
      style={
        Object {
          "marginBottom": "3%",
        }
      }
    >
      <img
        className="StyledImage-ey4zx9-0 iBeQsm"
        src="lock.svg"
      />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 gfocoz"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 jerVVw"
      >
        <div
          className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
        >
          <input
            autoComplete="off"
            className="StyledTextInput-sc-1x30a0s-0 hubMyb"
            name="userid"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Enter Registered ID"
            type="text"
          />
        </div>
        <button
          aria-label="Open Drop"
          className="StyledButton-sc-323bzc-0 ebqXLG Select__StyledSelectDropButton-sc-17idtfo-1 iGEQF"
          id="select"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 jmuHez"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 iAryDB"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
              >
                <input
                  autoComplete="off"
                  className="StyledTextInput-sc-1x30a0s-0 iuIygK Select__SelectTextInput-sc-17idtfo-0 bIurki"
                  id="select__input"
                  name="id"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Choose.."
                  readOnly={true}
                  tabIndex="-1"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="StyledBox-sc-13pk1d4-0 eA-DBEo"
              style={
                Object {
                  "minWidth": "auto",
                }
              }
            >
              <svg
                aria-label="FormDown"
                className="StyledIcon-ofa7kd-0 fuviCN"
                viewBox="0 0 24 24"
              >
                <polyline
                  fill="none"
                  points="18 9 12 15 6 9"
                  stroke="#000"
                  strokeWidth="2"
                />
              </svg>
            </div>
          </div>
        </button>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 cgsQvy"
      >
        <button
          className="StyledButton-sc-323bzc-0 iVPikU"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="submit"
        >
          Revoke
        </button>
      </div>
    </div>
  </div>
</div>
`;
